src/intel_pmu.c

```diff
@@ -297,17 +320,30 @@ static void pmu_dispatch_data(void) {
 
       event_enabled++;
 
+      /* If there are more events than counters, the kernel uses time
+       * multiplexing. With multiplexing, at the end of the run,
+       * the counter is scaled basing on total time enabled vs time running.
+       * final_count = raw_count * time_enabled/time_running
+       */
       uint64_t value = event_scaled_value(e, i);
       all_value += value;
 
+      /* get meta data with information about scaling */
+      meta_data_t *meta = pmu_event_get_meta(e, i);
+
       /* dispatch per CPU value */
-      pmu_submit_counter(i, e->event, value);
+      pmu_submit_counter(i, e->event, value, meta);
+
+      if (meta) {
```

Comment:
the `meta_data_destroy` handles the null ptr, so the `if` statement is not required here.

Reply:
Just checking that you saw this comment: there is no need to check if `meta` is non-NULL and imho the code is cleaner without the check, but the decision is up to you.

(A good practice is to reply something along the lines of "ACK; I prefer to keep as-is" to optional comments to show that you considered them.)

Reply:
Ok
Wanted to keep as-is but if two reviewers say the code will be cleaner I agree :)